sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3.java

```diff
@@ -531,11 +531,17 @@ private RunQueryRequest makeRequest(Query query) {
 
     /**
      * Datastore system tables with statistics are periodically updated. This method fetches
-     * the latest timestamp of statistics update using the {@code __Stat_Total__} table.
+     * the latest timestamp of statistics update using the {@code __Stat_Total__} or
+     * {@code __Stat_Ns_Total__} table.
      */
     private long queryLatestStatisticsTimestamp(Datastore datastore) throws DatastoreException {
       Query.Builder query = Query.newBuilder();
-      query.addKindBuilder().setName("__Stat_Total__");
```

Comment:
__Stat_Total__ kind is only available for "default" namespace. I tried 'GQL console' and it wasn't available for other namespaces. __Stat_Ns_Total works for other namespaces and returns the same timestamp (for the ones I checked). We could just skip the namespace altogether and just use '__Stat_Total__' but not sure if it would give the correct timestamp for all namespaces.

Skipping the timestamp do you mean something like `"select * from __Stat_Kind__ orderby 'timestamp' descending limit 1" `?